Goedi02 secure defaults (#62)
xsleaks/wiki: content/docs/defenses/opt-in/same-site-cookies.md

menu = "main"
+++

> Same-site cookies are one of the most impactful modern security mechanisms for fixing security issues that involve cross-site requests. This mechanism allows applications to force browsers to only include cookies in requests that are issued same-site [^1]. This type of cookie has three modes: `None`, `Lax`, and `Strict`.

Reviewer: ```suggestion
SameSite cookies are one of the most impactful modern security mechanisms for fixing security issues that involve cross-site requests. This mechanism allows applications to force browsers to only include cookies in requests that are issued same-site [^1]. This type of cookie has three modes: `None`, `Lax`, and `Strict`.
```

Looks like Same-site cookies are referred as SameSite cookies, but without dash:
e.g. https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie/SameSite so I guess we can follow the trend

## Same-Site Cookie Modes

> The following same-site cookie modes are available:

Reviewer: ```suggestion
The following SameSite cookie modes are available:
```

* `None` – Disables all protections and restores the old behavior of cookies. This mode is not recommended.

    {{< hint warning >}}
The `None` attribute must be accompanied by the `Secure` flag [^same-site-none].
[^same-site-none]: SameSite cookies explained, [link](https://web.dev/samesite-cookies-explained/#samesitenone-must-be-secure)
{{< /hint >}}

> * `Strict` – Causes the browser to not include cookies in any cross-site requests. This means `<script src="example.com/resource">`, `<img src="example.com/resource">`, `fetch()`, and `XHR` will all make requests without the same-site `Strict` cookies attached. Even if the user clicks on a link to `example.com/resource`, their cookies are not included.

Reviewer: ```suggestion
* `Strict` – Causes the browser to not include cookies in any cross-site requests. This means `<script src="example.com/resource">`, `<img src="example.com/resource">`, `fetch()`, and `XHR` will all make requests without the SameSite `Strict` cookies attached. Even if the user clicks on a link to `example.com/resource`, their cookies are not included.
```

* `Lax` – The only difference between `Lax` and `Strict` is that `Lax` mode allows cookies to be added to requests triggered by top-level navigations. This makes `Lax` cookies much easier to deploy since they won't break incoming links to your application. Unfortunately, an attacker can trigger a top-level navigation via `window.open` that allows the attacker to maintain a reference to the `window` object.

## Considerations

`Strict` cookies provide the strongest security guarantees, but it can be very difficult to deploy `Strict` same-site cookies in an existing application.

> Same-Site cookies are neither bulletproof [^2] nor can they fix everything. To complete the defense strategy against XS-Leaks, users should consider implementing other protections that complement same-site cookies. For example, [COOP]({{< ref "coop.md" >}}) can prevent an attacker from controlling pages using a `window` reference even if `Lax` same-site cookies are used.

Reviewer: ```suggestion
SameSite cookies are neither bulletproof [^2] nor can they fix everything. To complement the defense strategy against XS-Leaks, applications should consider implementing other protections besides, for example, [COOP]({{< ref "coop.md" >}}) can prevent an attacker from controlling pages using a `window` reference even if SameSite cookies in `Lax` mode are used.
```

`Strict` cookies provide the strongest security guarantees, but it can be very difficult to deploy `Strict` same-site cookies in an existing application.

> SameSite cookies are neither bulletproof [^2] nor can they fix everything. To complement the defense strategy against XS-Leaks, applications should consider implementing other, additional protections. For example, [COOP]({{< ref "coop.md" >}}) can prevent an attacker from controlling pages using a `window` reference even if SameSite cookies in `Lax` mode are used.

Reviewer: ```suggestion
SameSite cookies are neither bulletproof [^2] nor can they fix everything. To complement this defense strategy against XS-Leaks, applications should consider implementing other, additional protections. For example, [COOP]({{< ref "coop.md" >}}) can prevent an attacker from controlling pages using a `window` reference even if SameSite cookies in `Lax` mode are used.
```

One last small NIT to clarify what defense mechanism we have in mind.
